test/unit/java.test.ts

```diff
@@ -101,36 +111,131 @@ describe('java', () => {
 
   describe('version should be detected correctly', () => {
     it('the SonarQube version should be fetched correctly when new endpoint does not exist', async () => {
-      serverHandler.mockServerErrorResponse();
-      serverHandler.mockServerVersionResponse('3.2.2');
+      const token = 'dummy-token';
+      mock.onGet('http://sonarqube.com/api/server/version').reply(config => {
+        console.log({ config });
+        return [200, '3.2.2'];
+      });
```

Comment:
Discussion -> I'm a bit worried about losing the server/endpoint abstraction, as we're repeating a lot of these verbose mocks and it'll get worse when we create end-to-end tests for the `scan.ts`. For these tests, having an abstraction like
`xxx.mockServerVersionResponse('3.2.2');`

will be easier to maintain than repeating lines for each test such as 
`mock.onGet('http://sonarqube.com/api/server/version').reply(config => { return [200, '3.2.2']; })`

IMO, that was the rationale behind `ServerMock.ts` - I think it's fine for this PR, but when we're adding tests for `scan.ts` I'd argue that will be needed, probably we can discuss this later when we come to it

Reply:
I see what you mean, I think a ServerMock.ts class makes sense if the requests are very complex, At the same time you could argue it helps encapsulate the mock logic to keep the tests clean.

I usually start with simple inline approach and then decide when to make the abstraction. I'm fine with either approach here and we can bring back the `ServerMock.ts`